open_in_mpv/open_in_mpv.py

```diff
@@ -95,6 +112,68 @@ def remove_socket() -> bool:
     return True
 
 
+def mpv_executable() -> str:
+    return 'mpv.exe' if IS_WIN else 'mpv'
+
+
+@callback
+def mpv_launch(url: str, debug: bool, log: TextIO, new_env: Mapping[str, str]) -> None:
+    sp.check_call((
+        mpv_executable(),
+        '--gpu-api=opengl',
+        '--player-operation-mode=pseudo-gui',
+        '--quiet',
+        f'--input-ipc-server={MPV_SOCKET}',
+        url,
+    ) + ((f'--log-file={log.name}',) if debug else ()),
+                    env=new_env,
+                    stderr=log,
+                    stdout=log)
+    if not remove_socket():
+        logger.error('Failed to remove socket file.')
+
+
+@callback
+def mpv_launch_socket(url: str, debug: bool, log: TextIO, new_env: Mapping[str, str]) -> None:
+    if IS_WIN:
+        try:
+            handle = CreateFile(MPV_SOCKET, GENERIC_READ | GENERIC_WRITE,
+                                0, None, OPEN_EXISTING, 0, None)
+
+            #cspell:disable-next-line
+            if (res :=  SetNamedPipeHandleState(handle, PIPE_READMODE_MESSAGE, None, None)) == 0:
+                logger.debug(f'SetNamedPipeHandleState return code: {res}')
+                CALLBACKS['mpv_and_cleanup'](url, debug, log, new_env)
+                return
+
+            WriteFile(handle, json.dumps(dict(command=['loadfile', url],
+                                              request_id=random.randint(0x01, 0x7fffffff))
```

Comment:
What is `request_id` for? It doesn't happen on Linux/macOS.

Reply:
This was based on the implementation i found in C# for loadfile using Windows pipes; i'm not sure exactly my self as to if this is an mpv thing or a windows thing itself; ill try it both ways on windows and if it can work without it i will remove it, otherwise it will have to stay and differ from the Unix variant.